models/bancor/ethereum/bancor_v1_ethereum_trades.sql

```diff
@@ -0,0 +1,177 @@
+{{ config(
+    schema = 'bancor_v1_ethereum',
+    alias = 'trades',
+    partition_by = ['block_date'],
+    materialized = 'incremental',
+    file_format = 'delta',
+    incremental_strategy = 'merge',
+    unique_key = ['block_date', 'unique_trade_id'],
+    post_hook='{{ expose_spells(\'["ethereum"]\',
+                                "project",
+                                "bancor_v1",
+                                \'["codingsh"]\') }}'
+    )
+}}
+WITH dexs AS
+(
+    -- Bancor Network
+    SELECT
+        block_time,
+        'Bancor Network' AS project,
+        version::text AS version,
+        'DEX' AS category,
+        trader AS trader_a,
+        NULL::bytea AS trader_b,
+        target_token_amount_raw AS token_a_amount_raw,
+        source_token_amount_raw AS token_b_amount_raw,
+        NULL::numeric AS usd_amount,
+        CASE WHEN target_token_address = '\xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE' THEN
+            '\xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2'::bytea
+        ELSE target_token_address
+        END AS token_a_address,
+        CASE WHEN source_token_address = '\xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE' THEN
+            '\xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2'::bytea
+        ELSE source_token_address
+        END AS token_b_address,
+        contract_address AS exchange_contract_address,
+        tx_hash,
+        NULL::integer[] AS trace_address,
```

Comment:
So I see there is a runtime error about `NULL::integer[]` in https://github.com/duneanalytics/spellbook/actions/runs/3170613995/jobs/5163304788
maybe you can replace it with `CAST(NULL AS ARRAY<INT>)`
You can try to run `dbt compile` locally and test the compiled sql on Dune for testing.

Reply:
if you look at other dex scripts, we are just hardcoding blank space ```''``` for ```trace_address``` as none of the models were using the field at the time